docs/docs/overview/how_it_works.md

```diff
@@ -6,34 +6,44 @@ description: Secretless Broker Documentation
+1. Secretless Broker uses the secrets to open a connection to the Target Service.
 1. Secretless Broker pipes traffic between the Client and the Target Service.
-1. If a secret is changed, the Broker automatically obtains out the new secret and uses it to establish new connections.
+1. If a secret is changed, the Broker automatically obtains the new secret and uses it when establishing new connections.
 
 
-The Secretless Broker is a proxy which intercepts traffic to the backend service and performs the authentication phase of the backend protocol. The data-transfer phases of the protocol are direct pass-through between the client and backend service.
+The Secretless Broker is a proxy that intercepts traffic to the Target Service and performs the authentication phase of the backend protocol. The data-transfer phases of the protocol are direct pass-through between the client and Target Service.
 
-Examples of protocols which can be brokered:  
+Examples of protocols that can be brokered:  
 
 -  Database protocols such as Oracle, Postgresql, MySQL, NoSQL flavors, etc.
 -  HTTP via Authorization header
 -  SSH, via MITM or by implementing an ssh-agent   
 
-Any published protocol can be supported in the Secretless Broker. Software code in the Secretless Broker is generally required for each new protocol. See <a href="/docs/reference/handlers/overview.html">Handlers</a>.
+Any published protocol can be supported in the Secretless Broker. Software code in the Secretless Broker is generally required for each new protocol. For a list of currently supported Target Services, please see our documentation on <a href="/docs/reference/handlers/overview.html">Handlers</a>.
 
-<!-- TODO - link to handler section of reference page -->
+The Secretless Broker typically runs locally alongside the client application. Authentication between the Client and the Secretless Broker is managed by the operating system, e.g. local connection via Unix socket or HTTP connection to 127.0.0.1.  In container-managed environments such as Kubernetes, the Secretless Broker can be a “sidecar” container which is securely networked to the application container.
 
-The Connection Broker typically runs locally alongside the client application. Authentication between the Client and the Connection Broker is managed by the operating system, e.g. local connection via Unix socket or HTTP connection to 127.0.0.1.  In container-managed environments such as Kubernetes, the Connection Broker can be a “sidecar” container which is securely networked to the application container.
+
+## Internal Architecture
+
+<img src="/img/secretless_internal_architecture.svg" alt="Secretless Broker Internal Architecture">
+
+Internally, when the Secretless Broker receives a new connection request:
+1. The Proxy determines the Listener to send the request to, based on the port / socket the request came in on (there is a Listener for each potential connection to a Target Service)
```

Comment:
"based on request port / socket" ? 
"... the request came in on" has an odd ring to it for me